tensorflow/compiler/jit/extract_outside_compilation_pass.cc

```diff
@@ -1803,8 +1805,9 @@ TF_ATTRIBUTE_NOINLINE Status ExtractOutsideCompilationForFuncCallNode(
     if (e->IsControlEdge()) {
       continue;
     }
-
-    TF_RET_CHECK(e->dst_input() >= 0 && e->dst_input() < inputs.size());
+    
+    const int input_size_check = e->dst_input() < inputs.size(); 
+    TF_RET_CHECK(e->dst_input() >= 0 && input_size_check);
```

Comment:
I am unsure about this change. What is the warning that this handles?

Reply:
````
tensorflow/compiler/jit/extract_outside_compilation_pass.cc:1807:56: warning: comparison of integer expressions of different signedness: 'int' and 'std::vector<tensorflow::NodeDefBuilder::NodeOut>::size_type' {aka 'long unsigned int'} [-Wsign-compare]
 1807 |     TF_RET_CHECK(e->dst_input() >= 0 && e->dst_input() < inputs.size());
      |                                         ~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~
```

Reply:
But it will be the same warning if you extract the exact same comparison above. I think you need a `static_cast` on one of the terms. Then the `input_size_check` type should be `bool`